tests/includes/charmcraft.sh

```diff
@@ -0,0 +1,12 @@
+# pack_charm uses charmcraft to pack the local charm at the given directory,
+# and returns a path to the packed charm which can be supplied to juju deploy.
+#
+# Example usage:
+#    juju deploy $(pack_charm ./testcharms/charms/lxd-profile)
+pack_charm() {
+	local CHARM_DIR=$1
+	CHARM_NAME=$(basename "$CHARM_DIR")
+
+	charmcraft pack -p "$CHARM_DIR"
+	echo "./${CHARM_NAME}_*.charm"
+}
```

Comment:
It would be really great if you could cache charms on the host file system. That way we can save a bunch of time when running tests locally

Reply:
Charmcraft internally caches a lot of stuff, so packing charms is way quicker on subsequent runs, like 15 seconds.

Reply:
Yeah fair

Reply:
@barrettj12 is correct here.

Do we know if it uses lxd by default, if not, it might be worth testing that using `charmcraft pack --use-lxd`. If it is, just ignore me :)

Reply:
Yes, it uses LXD by default. You have to pass `--destructive-mode` to build on the host.